src/utilities/poisson.jl

```diff
@@ -9,6 +9,11 @@ timestep of simulation.
 - `l`: array of values
 """
 
-function poissonST(l::AbstractVector)
-    return poissonST(t) = [Float64(rand(Bernoulli(p), 1)[1]) for p in normalize(l)]
+function getPoissonST_old(l::AbstractVector)
+    return (_) -> rand.(Bernoulli.(normalize(l)))
+end
+
+function getPoissonST(l::AbstractVector, d)
+    norm = normalize(l)
+    return (_)->rand.(d.(norm))
```

Comment:
I think it'd be good to have this turn your input distribution `d` into a `product_distribution` so you don't have to broadcast the `rand` call. That's a bit clearer to read and understand than broadcasting `rand`. 

Once you get it over to a `product_distribution` I can run the tests locally and get it merged with `master`.